### ISSUE TITLE
Overrides of dependencies dependencies

### ISSUE THREAD

**Issue by ben657** [jspm/jspm-cli#1979]
So in our project we're writing all of our packages using jspm to manage dependencies, trying to keep them all as low down the tree as possible. For example:

Package A depends on B.
B depends on C.
So package A doesn't directly depend on C.
B needs to override a couple of properties in C, and does so in its package.json.
A now needs to know this so that when it imports B, B can import C properly.

Right now I don't think there's any way to do this without installing C as a dependency of A right? If so, shouldn't there be? Stopping us from really getting the most out of jspm I think.

**Comment by guybedford** [comment]
@ben657 the reason for this is to encourage overrides being checked into the jspm registry.

If everyone uses custom overrides, we end up with no common configuration, which is one of the main ideas in jspm and why transitive overrides aren't supported (from an implementation perspective this is easy to add).

Perhaps I'm being too harsh here, so if someone has a really strong argument for why we should support transitive overrides then we could support them :)

**Comment by guybedford** [comment]
To clarify the argument - as a user you want to naturally configure the things you install.

As a library publisher, you then want to naturally publish your custom configurations.

This is a bad path, because you've ignored the fact that to have public code with the rest of the world, your configurations need to match the configurations in the public package space.

If all packages started doing this, we end up in a position where we need to resolve override "conflicts"...

**Comment by guybedford** [comment]
Short version - is there a reason you can't check your overrides into the public jspm registry?

**Comment by ben657** [comment]
I see where you're coming from now, I can understand that we want everyone to contribute and have the majority of packages just working as they're installed. 

But there are times when you'll want small changes, or even to just get the change in quickly to test without needing to wait for a pull request to be approved on the registry.. not that i'm saying you're slow with that, just slower than hitting save and re-linking.

Basically, it's not that I _can't_ check my overrides in, it's just faster and less dependent on other people having to be there to use local overrides first. Maybe even some way of doing temporary transitive overrides until a pull is accepted would work?

**Comment by ben657** [comment]
I do understand the problem with override conflicts though, although I guess it then becomes up to the library publisher to maintain his overrides or get rid of them and use the registry, which personally I don't see as a bad thing.

**Comment by guybedford** [comment]
@ben657 the problem comes if it starts to become a pattern seeing `override` in the package.json and users assume that is the standard way and no one checks into the registry. Then the whole jspm package ecosystem becomes an ecosystem of conflicting overrides. So the idea is to try to never go down that path.

Having local overrides is exactly the quick development path.

The original idea was to have enough registry maintainers that all overrides can be reviewed and accepted within 5 minutes. I try to be quick, but unfortunately no one else is helping to review despite my adding people.

**Comment by ben657** [comment]
@guybedford I think I get what you're saying a bit more now, not having conflicts at all is probably a far better idea than forcing people to maintain and fix them.

I've actually just read up on using a private registry, I think that solves my problem. Easy enough to check in overrides to that, then myself and the team can use them immediately, whilst I get around to making a pr. Will give that workflow a try for now.

**Comment by guybedford** [comment]
Great, glad to hear it makes sense.
